src/core/core/rpc_message.cpp

```diff
@@ -266,6 +290,9 @@ void message_ex::seal(bool fill_crc)
     dassert  (!_is_read && _rw_committed, "seal can only be applied to write mode messages");
     //dbg_dassert(header->body_length > 0, "message %s is empty!", header->rpc_name);
 
+    if (hdr_format != NET_HDR_DSN)
```

Comment:
It's useless work to calculate crc for non-dsn header, because the crc will not be transmitted to the remote.

Actually, I think it is better to put calculation and checking of header and body crc in message_parser implement, but not in ***message_ex***. For different kind of header, the crc is different.

For example, for thrift header, the header crc should be calculated on ***thrift_message_header*** but not on ***message_header***; And, for http header, maybe only body crc is needed.